packages/cli/src/models/initializer/truffle/Truffle.ts

```diff
@@ -52,6 +52,21 @@ const Truffle = {
     }
   },
 
+  getContractNames(): string[] {
+    const buildDir = this.getBuildDir();
+    if (buildDir) {
+      return FileSystem.readDir(buildDir)
+        .filter((name) => {
+          const contract = FileSystem.parseJson(`${buildDir}/${name}`);
+          const projectDir = buildDir.replace('build/contracts', '');
+          return contract.bytecode.length > 2 && contract.sourcePath.match(projectDir, 'g') !== null;
```

Comment:
Note that `match` will turn `projectDir` into a regex, which may not be what we want here - maybe an `indexOf` is a better option.